htlcswitch/switch.go

```diff
@@ -142,6 +141,10 @@ type Config struct {
 	// provide payment senders our latest policy when sending encrypted
 	// error messages.
 	FetchLastChannelUpdate func(lnwire.ShortChannelID) (*lnwire.ChannelUpdate, error)
+
+	// Notifier is an instance of a chain notifier that we'll use to signal
+	// the switch when a new block has arrived.
+	Notifier chainntnfs.ChainNotifier
```

Comment:
Why pass the full notifier, when we can pass an already instantiated `BlockEpoch` struct?

Reply:
By the time the switch is declared in `server.go`, the `ChainNotifier` instance hasn't been initialized yet.